### DISCUSSION TITLE
成為googler的方法

### DISCUSSION THREAD

**Discussion by Doreen331** [1101-GS4719-C/Python-Community-GS4719-C#949] [General]
文章裡除了分享如何刷題還有一些資源，大家可以點進去看看～
但我想分享的是刷題這件事，刷題好像是變強或是找到工作的一條必經之路，之前在網路上看過另一個轉職的大神分享，他是在大四的時候決定往cs發展，當時他甚至不知道binary search是什麼，但之後花了一年的時間刷題，慢慢變強之後，現在在灣區年薪已經三千萬台幣了，但刷題之前還是要先把基本功學好！
https://vocus.cc/article/63b195e6fd89780001da4619

**Comment by Evan0979** [comment]
真的刷題之前還是要先把基本功學好,不然很容易就只學會一些片面的東西!!

**Comment by BlahFish1006** [comment]
感謝世界上所有不藏私的大神們的分享。
真的不管是哪科，掌握基本知識之後，下一步就得靠瘋狂刷題練習才會清楚知道自己哪裡不足、需要再加強，並且越來越熟悉使用。

**Comment by Tuoresa** [comment]
太強大了！！！要達到目標之前懂得獲取資源真的是非常重要的一個能力！
感謝分享的大神也感謝這位分享的同學哈哈
感覺對於程式學習刷題是必須經歷的過程，就算已經學習到基本語法，
但想要靈活運用感覺還是必須透過面對與解決各種不同情況，去提升自己的思考敏感度。

**Comment by woohsin** [comment]
刷題確實是許多人在學習程式設計或是準備面試時的必經之路。通過解題，不僅可以提升自己的演算法和編程能力，還能夠熟悉各種類型的面試題目，提高應對面試的能力。

像你提到的那位大神一樣，不論是對於轉行還是就業，刷題都是一個很有效的方法。當然，刷題的過程中，建立起良好的學習習慣和基本功是非常重要的。只有具備了堅實的基礎，才能更好地應對各種複雜的問題，並找到解決方案。

另外，刷題的同時也要注重理解和思考，而不僅僅是追求解題的數量。理解問題的本質、掌握解題的思路和方法，對於提升自己的能力更為重要。在這個過程中，也可以通過參考他人的解題思路和討論，來擴展自己的思維。

總的來說，刷題是一個良好的學習和成長方式，但同時也要注重方法和質量，並且不斷地提升自己的基本功和思考能力。

**Comment by jylin34** [comment]
白板題自從某一年在Google面試中出現後，現在已經成為軟體工程師面試的必備考題。白板題能夠檢驗一個人的智力、思考方式以及對演算法和資料結構的熟悉程度。然而，在實際的軟體工程中，成為一名合格的軟體工程師需要的技能遠不止這些。我相信，大部分時間並不會遇到需要白板題中所涉及的演算法或資料結構。因此，在進入大公司之前，刷題固然是必經之路，但學習其他技能同樣重要。

**Comment by NcuJerryKao** [comment]
建立基本功：在開始刷題前，確保自己對基本的數據結構和算法有紮實的理解，例如鏈表、二叉樹、圖算法、排序和搜索等。這些基本功是進一步提升的基礎 。

選擇合適的題目：針對自己的弱點和面試公司的常見題型，選擇相應的題目進行練習。LeetCode、HackerRank等平台提供了各種不同難度和類型的題目，可以按需選擇 。

定期複習：刷題並不只是單純解題，還需要定期回顧之前的錯題和難題，理解錯誤的原因並鞏固相應的知識點。這樣可以避免重複犯同樣的錯誤 。

模擬面試：在掌握了大量題目後，進行模擬面試練習，提升自己在壓力下解題的能力和表達的清晰度。可以邀請朋友幫忙或使用專業的模擬面試平台 。

尋求社群支持：參加相關的學習社群或討論區，和其他刷題者交流經驗、分享心得，互相激勵和幫助。這樣可以獲得更多的學習資源和技巧 。

**Comment by LinChloe** [comment]
確實刷題後會時常發現自己的知識不足。而且刷題需要人平靜地坐下來並且長時間有耐心地執行，這部分就會淘汰一堆人。
總而言之，敬佩大神，感謝大神，模仿大神，成為大神!!!

**Comment by hbh10601** [comment]
讀大學到現在越學越覺得每個人真的都有自己的長處，往自己最想要、最有興趣發展的方向才能學的持久！

**Comment by SusuTima** [comment]
感謝版主的分享，刷題很重要，但是弄懂每個題目背後的解題思路更為重要，目前也還是一個刷題小白，希望未來能多多加油。

**Comment by Shang621** [comment]
要成為Google的一員，首先要精通相關技術和領域知識，並擁有相關的學術背景或工作經驗。其次，積極參與開源項目、技術社區和相關活動，建立良好的職業聯繫和口碑。另外，準備充足的面試，包括技術面試和行為面試，並展示自己的解決問題能力和團隊合作精神。此外，不斷學習和成長，保持對技術的熱情和創新思維。最後，通過Google的招聘流程，包括線上申請、面試和審核，並展示自己的價值和適應能力。

**Comment by cassy0201** [comment]
我覺得不管什麼科目都能靠刷題累積經驗，同時也能了解自己不足，針對性的改進!

**Comment by 710388** [comment]
不斷的學習跟練習題目吧

**Comment by yubo0oo** [comment]
寫越多題目才會越發現自己的不足之處
人外有人天外有天
